--- a/plugin/down-root/Makefile	2012-08-25 15:32:43.000000000 +0800
+++ b/plugin/down-root/Makefile	2012-08-25 15:36:02.000000000 +0800
@@ -8,10 +8,10 @@
 CC_FLAGS=-O2 -Wall
 
 down-root.so : down-root.o
-	gcc ${CC_FLAGS} -fPIC -shared -Wl,-soname,openvpn-down-root.so -o openvpn-down-root.so down-root.o -lc
+	$(CC) ${CC_FLAGS} -fPIC -bundle -Wl,-dylib_install_name,openvpn-down-root.so -o openvpn-down-root.so down-root.o -lc
 
 down-root.o : down-root.c
-	gcc ${CC_FLAGS} -fPIC -c ${INCLUDE} down-root.c
+	$(CC) ${CC_FLAGS} -fPIC -c ${INCLUDE} down-root.c
 
 clean :
 	rm -f *.o *.so
diff -ur syshead.h syshead.h
--- syshead.h   2011-06-23 23:13:39.000000000 -0700
+++ syshead.h   2011-07-26 14:49:07.000000000 -0700
@@ -392,6 +392,13 @@
 #endif
 
 /*
+ * Define a proper IP socket level if not already done.
+ */
+#ifndef SOL_IP
+#define SOL_IP IPPROTO_IP
+#endif
+
+/*
  * Disable ESEC
  */
 #if 0

